lib/ngx/ssl/session.md

```diff
@@ -0,0 +1,277 @@
+Name
+====
+
+ngx.ssl.session - Lua API for manipulating SSL session data and IDs for NGINX downstream SSL connections.
+
+Table of Contents
+=================
+
+* [Name](#name)
+* [Status](#status)
+* [Synopsis](#synopsis)
+* [Description](#description)
+* [Methods](#methods)
+    * [get_session_id](#get_session_id)
+    * [get_serialized_session](#get_serialized_session)
+    * [set_serialized_session](#set_serialized_session)
+* [Community](#community)
+    * [English Mailing List](#english-mailing-list)
+    * [Chinese Mailing List](#chinese-mailing-list)
+* [Bugs and Patches](#bugs-and-patches)
+* [Author](#author)
+* [Copyright and License](#copyright-and-license)
+* [See Also](#see-also)
+
+Status
+======
+
+This Lua module is currently considered experimental.
+
+Synopsis
+========
+
+```nginx
+# nginx.conf
+
+# Note: you do not need the following line if you are using
+# OpenResty 1.11.2.1+.
+lua_package_path "/path/to/lua-resty-core/lib/?.lua;;";
+
+server {
+    listen 443 ssl;
+    server_name test.com;
+
+    # well, we could configure ssl_certificate_by_lua* here as well...
+    ssl_certificate /path/to/server-cert.pem;
+    ssl_certificate_key /path/to/server-priv-key.pem;
+
+    ssl_session_fetch_by_lua_block {
+        local ssl_sess = require "ngx.ssl.session"
+
+        local sess_id, err = ssl_sess.get_session_id()
+        if not sess_id then
+            ngx.log(ngx.ERR, "failed to get session ID: ", err)
+            -- considered a cache miss, and just return...
+            return
+        end
+
+        -- the user is supposed to implement the my_lookup_ssl_session_by_id
+        -- Lua function used below. She can look up an external memcached
+        -- or redis cluster, for example. And she can also introduce a local
+        -- cache layer at the same time...
```

Comment:
IMHO, the above comment is not very straightforward, I would say "It is up to the user to implement xyz() function, which is to retrieve the session corresponding to the given id".